### PR TITLE
Make driver version consistent in ODBC Connection

### PR DESCRIPTION
**Pull request by Xtema**
Update example code to be aligned and in sync (version used in the example)

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by dirrao — APPROVED** [review]

**Review by RNHTTR — APPROVED** [review]

**Review by Taragolis — APPROVED** [review]

**Comment by Taragolis** [comment]
@Xtema you don't need to update this PR every time as main updated, since it is your first commit every time I need to re-approve workflow again and wait until it pass CI before I could merge it into the `main`

**Pull request merged**
